src/features/messages/MessageSearch.tsx

```diff
@@ -138,16 +139,29 @@ export function MessageSearch() {
           <h2 className="w-min pl-0.5 font-medium text-blue-500 sm:w-fit">
             {!hasInput ? 'Latest Messages' : 'Search Results'}
           </h2>
-          <SearchFilterBar
-            originChain={originChainFilter}
-            onChangeOrigin={setOriginChainFilter}
-            destinationChain={destinationChainFilter}
-            onChangeDestination={setDestinationChainFilter}
-            startTimestamp={startTimeFilter}
-            onChangeStartTimestamp={setStartTimeFilter}
-            endTimestamp={endTimeFilter}
-            onChangeEndTimestamp={setEndTimeFilter}
-          />
+          <div className="flex items-center space-x-2 md:space-x-4">
+            <SearchFilterBar
+              originChain={originChainFilter}
+              onChangeOrigin={setOriginChainFilter}
+              destinationChain={destinationChainFilter}
+              onChangeDestination={setDestinationChainFilter}
+              startTimestamp={startTimeFilter}
+              onChangeStartTimestamp={setStartTimeFilter}
+              endTimestamp={endTimeFilter}
+              onChangeEndTimestamp={setEndTimeFilter}
+            />
+            <IconButton
+              onClick={refetch}
+              className="h-7 w-7 rounded-lg bg-pink-500"
+              disabled={isAnyFetching}
+            >
+              {isAnyFetching ? (
+                <SpinnerIcon color="white" height={16} width={16} />
+              ) : (
+                <RefreshIcon color="white" height={20} width={20} />
+              )}
+            </IconButton>
```

Comment:
Nit: I'd slightly prefer this be in it's own component. It can be just lower in this file